driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -2282,6 +2289,234 @@ public void run() {
+            }
+        }
+
+        private class SchemaRefreshRequestDeliveryCallback implements EventDebouncer.DeliveryCallback<SchemaRefreshRequest> {
+
+            @Override
+            public ListenableFuture<?> deliver(final List<SchemaRefreshRequest> events) {
+                return executor.submit(new ExceptionCatchingRunnable() {
+                    @Override
+                    public void runMayThrow() throws InterruptedException, ExecutionException {
+                        SchemaRefreshRequest coalesced = null;
+                        for (SchemaRefreshRequest request : events) {
+                            coalesced = coalesced == null ? request : coalesced.coalesce(request);
+                        }
+                        assert coalesced != null;
+                        logger.trace("Coalesced schema refresh request: {}", coalesced);
+                        controlConnection.refreshSchema(coalesced.keyspace, coalesced.table);
+                    }
+                });
+            }
+
+        }
+
+        private class NodeRefreshRequest {
+
+            private final Host host;
+
+            private final HostEvent eventType;
+
+            private NodeRefreshRequest(Host host, HostEvent eventType) {
+                this.host = host;
+                this.eventType = eventType;
+            }
+
+            @Override
+            public String toString() {
+                return host + " " + eventType;
+            }
+
+        }
+
+        private class NodeRefreshRequestDeliveryCallback implements EventDebouncer.DeliveryCallback<NodeRefreshRequest> {
+
+            @Override
+            public ListenableFuture<?> deliver(List<NodeRefreshRequest> events) {
+                Map<Host, HostEvent> hosts = new HashMap<Host, HostEvent>();
+                // only keep the last event for each host
+                for (NodeRefreshRequest req : events) {
+                    hosts.put(req.host, req.eventType);
+                }
+                List<ListenableFuture<?>> futures = new ArrayList<ListenableFuture<?>>(hosts.size());
+                for (final Entry<Host, HostEvent> entry : hosts.entrySet()) {
+                    final Host host = entry.getKey();
+                    HostEvent eventType = entry.getValue();
+                    switch (eventType) {
+                        case UP:
+                            schedule(hostUp(host));
+                            break;
+                        case ADDED:
+                            schedule(hostAdded(host));
+                            break;
+                        case DOWN:
+                            execute(hostDown(host));
+                            break;
+                        case REMOVED:
+                            execute(hostRemoved(host));
+                            break;
+                    }
+                }
+                return Futures.allAsList(futures);
```

Comment:
It looks like futures is never populated, is that a problem?   It doesn't look like we ever wait on NodeRefreshEvents so it's probably not a problem at the moment.